util/testutil/shell.go

```diff
@@ -367,7 +367,7 @@ func KillMatchingProcess(sh *shell.Shell, psLinePattern string) error {
 
 	var allErr error
 	for _, pid := range targets {
-		if err := sh.Command("kill", "-9", fmt.Sprintf("%d", pid)).Run(); err != nil {
+		if err := sh.Command("kill", "-2", fmt.Sprintf("%d", pid)).Run(); err != nil {
```

Comment:
It might be worth a comment to explain why this needs to be SIGTERM instead of SIGKILL.

```suggestion
// Send SIGTERM so the unit under test correctly writes integration coverage reports to Go coverage directory.
```